src/cascadia/TerminalControl/ScreenInfoUiaProvider.hpp

```diff
@@ -0,0 +1,78 @@
+/*++
+Copyright (c) Microsoft Corporation
+Licensed under the MIT license.
+
+Module Name:
+- ScreenInfoUiaProvider.hpp
+
+Abstract:
+- This module provides UI Automation access to the screen buffer to
+  support both automation tests and accessibility (screen reading)
+  applications.
+- ConHost and Windows Terminal must use IRenderData to have access to the proper information
+- Based on examples, sample code, and guidance from
+  https://msdn.microsoft.com/en-us/library/windows/desktop/ee671596(v=vs.85).aspx
+
+Author(s):
+- Carlos Zamora   (CaZamor)   2019
+--*/
+
+#pragma once
+
+#include "..\types\ScreenInfoUiaProviderBase.h"
+#include "..\types\UiaTextRangeBase.hpp"
+#include "UiaTextRange.hpp"
+
+namespace Microsoft::Console::Render
+{
+    class IRenderData;
+}
+
+namespace Microsoft::Console::Types
+{
+    class WindowUiaProviderBase;
+}
+
+namespace Microsoft::Terminal
+{
+    class ScreenInfoUiaProvider : public Microsoft::Console::Types::ScreenInfoUiaProviderBase

```

Comment:
I'd kinda like it better if this was `TermControlUiaProvider`, esp if there's not really a `SCREEN_INFORMATION` in the TerminalControl.

Reply:
Just a renaming, right? That way we associate it more with the terminal and if we ever put a non-termcontrol in a pane, we dissacociate screen information from a terminal control.